App.tsx

```diff
@@ -0,0 +1,194 @@
+import React from 'react';
+import {
+  View,
+  Text,
+  TouchableOpacity,
+  Platform,
+  StyleSheet
+} from 'react-native';
+import NfcManager, {NfcTech} from 'react-native-nfc-manager';
+import {Buffer} from 'buffer';
+
+interface State {
+  apduResponse: string;
+  testOutput: boolean;
+  detectedTechnology: string;
+  nfcUID: string;
+  showInstructions: boolean;
+}
+
+export class App extends React.Component<any, State> {
+
+  constructor(props: any) {
+    super(props);
+    this.state = {
+      apduResponse: "Please run the test",
+      testOutput: false,
+      detectedTechnology: "",
+      nfcUID: "",
+      showInstructions: false
+    }
+  }
+
+  componentDidMount() {
+    NfcManager.start();
+  }
+
+  componentWillUnmount() {
+    this.cleanUp();
+  }
+
+  private cleanUp = () => {
+    NfcManager.cancelTechnologyRequest().catch(() => 0);
+    this.setState({
+      apduResponse: "Please run the test",
+      testOutput: false,
+      detectedTechnology: "",
+      nfcUID: "",
+      showInstructions: false
+    });
+  }
+
+  private test = async () => {
+    try {
+      let tech = NfcTech.IsoDep;
+
+      this.setState({showInstructions: true});
+
+      let resp = await NfcManager.requestTechnology(tech, {
+        alertMessage: 'Ready to send some APDU'
+      });
+
+      // the NFC uid can be found in tag.id
+      let tag = await NfcManager.getTag();
+
+
+      this.setState({
+        detectedTechnology: resp,
+        nfcUID: tag.id
+      });
+
+      // Test Case 7816_A_1
+      // Send the following SelectApplication APDU to the e-Passport.
+      // ‘00 A4 04 0C 07 A0 00 00 02 47 10 01'
+      // According to the ICAO recommendation, the P2 denotes "return no file
+      // information", and there is no Le byte present. Therefore, the response
+      // data field MUST be empty. The e-Passport MUST return status bytes 
+      //‘90 00’.
+
+      let apduAnswer = undefined;
+      if (Platform.OS === 'ios') {
+        apduAnswer = await NfcManager.sendCommandAPDUIOS([0x00, 0xA4, 0x04, 
+          0x0C, 0x07, 0xA0, 0x00, 0x00, 0x02, 0x47, 0x10, 0x01]);
+        /**
+         * or, you can use alternative form like this:
+           resp = await NfcManager.sendCommandAPDUIOS({
+             cla: 0,
+             ins: 0x84,
+             p1: 0,
+             p2: 0,
+             data: [],
+             le: 8
+           });
+         */
+      } else {
+        apduAnswer = await NfcManager.transceive([0x00, 0xA4, 0x04, 0x0C, 0x07, 
+          0xA0, 0x00, 0x00, 0x02, 0x47, 0x10, 0x01]);
+      }
+    
+      let humanReadableResponse = Buffer.from(apduAnswer).toString('hex');
+
+      this.setState({
+        apduResponse: humanReadableResponse,
+        testOutput: (humanReadableResponse === '9000'),
+        showInstructions: false}
+      );
+    } catch (ex) {
+      console.warn('Exception', ex);
```

Comment:
I guess it could be better to show the exception (as UI Text or as [Alert](https://reactnative.dev/docs/alert))

Reply:
Exception messages are somewhat foggy. Nevertheless, I'll show them in an Alert.